src/entity/Unit.ts

```diff
@@ -1444,6 +1450,91 @@ export function findLOSLocation(unit: IUnit, target: Vec2, underworld: Underworl
 
 }
 
+
+// handles drawing attack range, bloat radius, and similar graphics each frame while a unit is selected
+export async function drawSelectedGraphics(unit: IUnit, prediction: boolean = false, underworld: Underworld): Promise<void> {
+
+  if (!prediction) {
+    for (let drawEvent of unit.onDrawSelectedEvents) {
+      if (drawEvent) {
+        const fn = Events.onDrawSelectedSource[drawEvent];
+        if (fn) {
+          await fn(unit, prediction, underworld);
```

Comment:
Does this event function need to be async?

Reply:
Oh!... Probably not.